Signal/src/view controllers/TSMessageAdapter.m

```diff
@@ -237,6 +241,79 @@ - (NSDate *)date {
     return self.messageDate;
 }
 
+- (BOOL)canPerformEditingAction:(SEL)action {
+    if (action == @selector(delete:)) {
+        return YES;
+    } else if (action == @selector(copy:)) {
+        if (!self.isMediaMessage
+            || [self.mediaItem isKindOfClass:[TSPhotoAdapter class]]
+            || [self.mediaItem isKindOfClass:[TSAnimatedAdapter class]]) {
+
+            return YES;
+        }
+        //TODO implement copy for video
+    } else if (action == NSSelectorFromString(@"save:")) {
+        if ([self.mediaItem isKindOfClass:[TSPhotoAdapter class]]
+            ||[self.mediaItem isKindOfClass:[TSAnimatedAdapter class]]) {
+            return  YES;
+        }
+        //TODO implement save for video
+    }
+    return NO;
+}
+
+- (void)performEditingAction:(SEL)action {
+    if (action == @selector(delete:)) {
+        DDLogDebug(@"Deleting interaction with uniqueId: %@", self.interaction.uniqueId);
+        [self.interaction remove];
+    } else if (action == @selector(copy:)) {
```

Comment:
This is screaming for refactoring based on inheritance, but I was trying not to move too much at once.